### PR TITLE
[MSHARED-1128] Introduce execute method and deprecate executeGoal(s)

### PR DESCRIPTION
**Pull request by slawekjaranowski**

### CODE REVIEW

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 1]
```diff
@@ -1284,8 +1335,6 @@ public void executeGoals( List<String> goals, Map<String, String> envVars )
 
         if ( ret > 0 )
         {
-            System.err.println( "Exit code: " + ret );
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:1287]: Very good!

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 2]
```diff
@@ -1263,12 +1301,25 @@ public void executeGoals( List<String> goals, Map<String, String> envVars )
             args.add( "-Dmaven.repo.local=" + localRepo );
         }
 
-        args.addAll( allGoals );
+        if ( autoclean )
+        {
+            /*
+             * NOTE: Neither test lifecycle binding nor prefix resolution here but call the goal directly.
+             */
+            args.add( "org.apache.maven.plugins:maven-clean-plugin:clean" );
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:1309]: Would it make sense to move this value to a static final for documentation puposes?

**Review by michael-o — COMMENTED** [review]
A few comments left. What I have noticed is that both system properties and env vars are read-write and can be retrieved. What is your opinion on this, should they be read-only or not getter at all?

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java (hunk 2 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/verifier/Verifier.java]: ok, try with constant ....

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #155

**Pull request merged**